src/librustc_interface/passes.rs

```diff
@@ -267,15 +267,18 @@ fn configure_and_expand_inner<'a>(
             lint_store,
             &krate,
             true,
+            None,
             rustc_lint::BuiltinCombinedPreExpansionLintPass::new());
     });
 
+    let lint_buffer = lint::LintBuffer::default();
```

Comment:
`LintBuffer` doesn't contain references to `Resolver` fields, so it can be constructed inside `Resolver::new()`.